### PR TITLE
Check mEmail for empty String to prevent "credential identifier cannot be empty" error

### LINKED ISSUE #141: Facebook login error
I have gotten this error when using the new FirebaseUI with Facebook login:

```
java.lang.IllegalArgumentException: credential identifier cannot be empty
at com.firebase.ui.auth.ui.account_link.SaveCredentialsActivity.onConnected(SaveCredentialsActivity.java:120)
```

After inserting credentials and successfully logging in the app crashes and this is the error.

Do you know what might be going wrong?

### PR DESCRIPTION
**Pull request by stba42**
made code more robust, should fix #141

### CODE REVIEW

File: auth/src/main/java/com/firebase/ui/auth/ui/account_link/SaveCredentialsActivity.java [hunk 1]
```diff
@@ -89,7 +89,7 @@ protected void onStart() {
 
     @Override
     public void onConnected(@Nullable Bundle bundle) {
-        if (mEmail == null) {
+        if (mEmail == null || mEmail.equals("")) {
```
Comment (samtstern) [inline auth/src/main/java/com/firebase/ui/auth/ui/account_link/SaveCredentialsActivity.java]: This line can be replaced with `TextUtils.isEmpty(mEmail)`

**Comment by samtstern** [comment]
@iainmcgin please review.

File: auth/src/main/java/com/firebase/ui/auth/ui/account_link/SaveCredentialsActivity.java (hunk 1 above)
Comment (iainmcgin) [inline auth/src/main/java/com/firebase/ui/auth/ui/account_link/SaveCredentialsActivity.java]: We should investigate under what circumstances a null / empty email address ever arrives here. I agree with sam however that `if (TextUtils.isEmpty(mEmail)` is a more descriptive check here.

Comment (abidsoft) [inline auth/src/main/java/com/firebase/ui/auth/ui/account_link/SaveCredentialsActivity.java]: Empty email issue happens due to privacy setting of some users, where you get name and profile image but email is null

**Comment by SUPERCILEX** [comment]
This PR can be closed now: fixed in #357.

**Comment by samtstern** [comment]
#357 merged.

**Pull request closed without merging**